case-lib/pipeline.sh

```diff
@@ -7,35 +7,26 @@ func_pipeline_export()
         dlogi "Topology file name is not specified, unable to run command: ${BASH_SOURCE[-1]}" && exit 1
     fi
 
-    local tplg="$1" opt="" sofcard=${SOFCARD:-0} cmd=""
+    local tplg="$1" ignore="" sofcard=${SOFCARD:-0}
 
     # got tplg_file, verify file exist
     tplg_path=`func_lib_get_tplg_path "$tplg"`
     [[ "$?" != "0" ]] && dloge "No available topology for pipeline export" && exit 1
     dlogi "${BASH_SOURCE[-1]} will use topology $tplg_path to run the test case"
 
-    shift
     # create filter option string
-    if [ $# -ne 0 ]; then
-        opt="-f"
-        while [ $# -ne 0 ]
-        do
-            opt=$opt" $1"
-            shift
-        done
-    fi
+    local filter=$2
 
     # create block option string
     if [ ${#TPLG_IGNORE_LST[@]} -ne 0 ]; then
-        opt=$opt" -b"
         for key in ${!TPLG_IGNORE_LST[@]}
         do
             dlogi "Pipeline list to ignore is specified, will ignore '$key=${TPLG_IGNORE_LST[$key]}' in test case"
-            opt=$opt" $key:'${TPLG_IGNORE_LST[$key]}'"
+            ignore=$ignore" $key:'${TPLG_IGNORE_LST[$key]}'"
         done
     fi
 
-    cmd=$(echo sof-tplgreader.py $tplg_path $opt -s $sofcard -e)
+    cmd=$(echo "sof-tplgreader.py $tplg_path -f \"$filter\" -b \"$ignore\" -s $sofcard -e")
```

Comment:
Why not just the simpler:
```shell
cmd="sof-tplgreader.py $tplg_path -f \"$filter\" -b \"$ignore\" -s $sofcard -e"
```
Or probably better:
```shell
cmd="sof-tplgreader.py $tplg_path -f '$filter' -b '$ignore' -s $sofcard -e"
```

Reply:
Thanks, I will use the first suggestion, as the single quote in second one is wrong.

```
2020-04-21 05:42:24 UTC [COMMAND] sof-tplgreader.py /lib/firmware/intel/sof-tplg/sof-cml-rt1011-rt5682.tplg -f 'type:capture & id:1' -b ' pcm:'HDA Digital,Media Playback,DMIC16k'' -s 0 -e

2020-04-21 05:45:29 UTC [COMMAND] sof-tplgreader.py /lib/firmware/intel/sof-tplg/sof-cml-rt1011-rt5682.tplg -f "type:capture & id:0" -b " pcm:'HDA Digital,Media Playback,DMIC16k'" -s 0 -e
```

Reply:
Embedding quotes inside strings is really complex and nasty and so is `eval` and `ssh $host $cmd`. Complex commands with whitespace like this one should be generated into `mktemp` files and then the main script runs  these files. So much easier to understand and debug.

In this case it's not even clear to me why `$cmd` is needed at all, is this just for logging it? Anyway this is a bit off-topic, I understand this PR has a more limited scope.